app/assets/javascripts/calendars/guiders-appointments.es6

```diff
@@ -0,0 +1,207 @@
+        eventTextColor: '#fff',
+        resourceRender: (resourceObj, labelTds, bodyTds, view) => {
+          if (view.type === 'agendaDay') {
+            labelTds.html('');
+            $(`<div>${resourceObj.title}</div>`).prependTo(labelTds);
+          } else {
+            $('<span aria-hidden="true" class="glyphicon glyphicon-user" style="margin-right: 5px;"></span>').prependTo(
+              labelTds.find('.fc-cell-text')
+            );
+          }
+        },
+        eventRender: (event, element, view) => {
+          $(element).attr('id', event.id);
+
+          if (view.type === 'agendaDay') {
+            element.find('.fc-content').remove();
+          } else {
+            $('<span class="glyphicon glyphicon-phone-alt" aria-hidden="true" style="margin-right: 5px;"></span>').prependTo(
+              element.find('.fc-content')
+            );
+          }
+
+          if (event.hasChanged) {
+            event.backgroundColor = 'red';
+            event.borderColor = '#c00';
+          } else {
+            delete event.backgroundColor;
+            delete event.borderColor;
+          }
+        },
+        eventAfterRender: (event, element) => {
+          if (event.rendering === 'background') {
+            return;
+          }
+
+          var resource = el.fullCalendar('getResourceById', event.resourceId);
+
+          element.qtip({
+            content: {
+              text: `
+              <p>${event.start.format('HH:mm')} - ${event.end.format('HH:mm')}</p>
+              <p><span class="glyphicon glyphicon-phone-alt" aria-hidden="true"></span> ${event.title}</p>
+              <p><span class="glyphicon glyphicon-user" aria-hidden="true"></span> ${resource.name} </p>
+              `
+            }
+          });
+        },
+        eventDrop: (event, delta, revertFunc) => {
+          this.handleEventChange(event, revertFunc);
+        },
+        eventResize: (event, delta, revertFunc) => {
+          this.handleEventChange(event, revertFunc);
+        },
+        resources: '/guiders',
+        events: '/appointments?include_links=false'
+      }, config);
+
+      super(el, calendarConfig);
+
+      this.eventChanges = [];
+      this.actionPanel = $('[data-action-panel]');
+
+      this.setupUndo();
+    }
+
+    setupUndo() {
+      this.actionPanel.find('[data-action-panel-undo-all]').on('click', this.undoAllChanges.bind(this));
+      this.actionPanel.find('[data-action-panel-undo-one]').on('click', this.undoOneChange.bind(this));
+      this.actionPanel.find('[data-action-panel-save]').on('click', this.save.bind(this));
+    }
+
+    handleEventChange(event, revertFunc) {
+      event.hasChanged = true;
+
+      this.eventChanges.push({
+        eventObj: event,
+        revertFunc: revertFunc
+      });
+
+      this.$el.fullCalendar('rerenderEvents');
+
+      this.checkToShowActionPanel();
+    }
+
+    undoOneChange(evt) {
+      evt.preventDefault();
+
+      const event = this.eventChanges.pop();
```

Comment:
I know this was lifted from another branch but we should try and make declarations at the beginning of the function.

Also, does `const` feel right here? Seems more like a `let` maybe.